### PR TITLE
add `cupy.heaviside` api.

### PR DESCRIPTION
**Pull request by tom24d**
As per title.
Ref: #6078

### CODE REVIEW

File: cupy/_math/misc.py [hunk 1]
```diff
@@ -239,6 +239,36 @@ def clip(a, a_min=None, a_max=None, out=None):
...
+    out0 = in1;
+} else {
+    out0 = (in0 > 0);
+}
+'''
+_float_heaviside = _float_heaviside_head + " else " + _heaviside
+heaviside = _core.create_ufunc(
+    'cupy_heaviside',
+    ('bb->b', 'BB->B', 'hh->h', 'HH->H', 'ii->i', 'II->I',
+     'll->l', 'LL->L', 'qq->q', 'QQ->Q',
```
Comment (asi1024) [inline cupy/_math/misc.py:258]: numpy.heaviside always return floating values. Could you remove these combinations?
```py
>>> numpy.heaviside(0, 1)
1.0
```

File: tests/cupy_tests/math_tests/test_misc.py [hunk 2]
```diff
@@ -452,6 +452,22 @@ def test_interp_inf_to_nan(self, xp, dtype_y, dtype_x):
         fy = xp.asarray([0, 10], dtype=dtype_y)
         return xp.interp(x, fx, fy)
 
+    @testing.for_all_dtypes(name='dtype_2', no_bool=True, no_complex=True)
+    @testing.for_all_dtypes(name='dtype_1', no_bool=True, no_complex=True)
+    @testing.numpy_cupy_array_equal(type_check=False)
```
Comment (asi1024) [inline tests/cupy_tests/math_tests/test_misc.py:457]: Let's check return types.

File: cupy/_math/misc.py (hunk 1 above)
Comment (tom24d) [reply cupy/_math/misc.py]: Fixed in 99825e52b7dd86e1da31664a202c13a93803242f.

File: tests/cupy_tests/math_tests/test_misc.py (hunk 2 above)
Comment (tom24d) [reply tests/cupy_tests/math_tests/test_misc.py]: Fixed in 546002fada1ec43bfd58090ffb7afbda591c4731.

**Comment by tom24d** [comment]
Thanks for the review!
PTAL.

**Review by takagi — APPROVED** [review]
LGTM! Waiting for CI pass.

**Comment by takagi** [comment]
Thanks, @tom24d !

**Pull request merged**